src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/DefaultViewComponentHelper.cs

```diff
@@ -210,7 +216,25 @@ private ViewComponentDescriptor SelectComponent(Type componentType)
                     Resources.FormatViewComponent_IViewComponentFactory_ReturnedNull(descriptor.Type.FullName));
             }
 
-            return invoker.InvokeAsync(context);
+#pragma warning disable 0618
+            if (_diagnosticSource.IsEnabled("Microsoft.AspNet.Mvc.BeforeViewComponent"))
+            {
+                _diagnosticSource.Write(
+                    "Microsoft.AspNet.Mvc.BeforeViewComponent",
+                    new { actionDescriptor = _viewContext.ActionDescriptor, viewComponentContext = context });
+            }
+
+            var result = invoker.InvokeAsync(context);
```

Comment:
The reason I put it here was putting it in invoker might lead to duplication because we call `InvokeSyncCore` in two places. But If we want the result, we should put it there. I'll move it.